app/controllers/resources_controller.rb

```diff
@@ -38,33 +39,15 @@ def permitted_roles
     render json: Role.that_can(privilege, @resource).map {|r| r.id}
   end
 
-  # Implements the use case "check permission on some resource",
-  # either for the role specified by the query string, or for the
-  # +current_user+.
+  # Implements the use case "check MY permission on some resource", where "me" is defined as the +current_user+.
   def check_permission
     privilege = params[:privilege]
     raise ArgumentError, "privilege" unless privilege
 
-    if query_role.allowed_to?(privilege, @resource)
+    if current_role_from_param(:role).allowed_to?(privilege, @resource)
```

Comment:
As a user of this API I'd find it surprising that it doesn't respect `acting_as` like most other APIs.

Reply:
I still think this is more like a query than like the assumption of a role. Also, changing this to expect `acting_as` will require an API change, which feels beyond the scope of this work.